src/Dfc.CourseDirectory.FindACourseApi/Features/TLevels/TLevels.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dfc.CourseDirectory.Core.DataStore.CosmosDb;
+using Dfc.CourseDirectory.Core.DataStore.Sql;
+using Dfc.CourseDirectory.Core.DataStore.Sql.Queries;
+using MediatR;
+using CosmosQueries = Dfc.CourseDirectory.Core.DataStore.CosmosDb.Queries;
+
+namespace Dfc.CourseDirectory.FindACourseApi.Features.TLevels
+{
+    public class Query : IRequest<ViewModel>
+    {
+    }
+
+    public class ViewModel
+    {
+        public IReadOnlyCollection<TLevelDetailViewModel> TLevels { get; set; }
+    }
+
+    public class Handler : IRequestHandler<Query, ViewModel>
+    {
+        private readonly ISqlQueryDispatcher _sqlQueryDispatcher;
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+
+        public Handler(ISqlQueryDispatcher sqlQueryDispatcher, ICosmosDbQueryDispatcher cosmosDbQueryDispatcher)
+        {
+            _sqlQueryDispatcher = sqlQueryDispatcher;
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+        }
+
+        public async Task<ViewModel> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var tLevels = await _sqlQueryDispatcher.ExecuteQuery(new GetTLevels());
+
+            var providerIds = tLevels.Select(t => t.ProviderId).Distinct();
+
+            var getProviders = _cosmosDbQueryDispatcher.ExecuteQuery(
+                new CosmosQueries.GetProvidersByIds { ProviderIds = providerIds });
+
+            var getSqlProviders = _sqlQueryDispatcher.ExecuteQuery(
+                new GetProvidersByIds { ProviderIds = providerIds });
+
+            var getVenues = _sqlQueryDispatcher.ExecuteQuery(
```

Comment:
I wonder whether we should add `Venue` onto `TLevelLocation` at some point since all these queries are already going to that table and we often need to lookup the `Venue` for each location. Probably one for the future given the number of places that would need changing.

Reply:
That could be useful moving forward, but perhaps out of scope for this.